File: awg_wallpaper/lib/models/community_post.dart
import 'community_user.dart';

class CommunityPost {
  final int id;
  final String imageUrl;
  final String? thumbnailUrl;
  final String? title;
  final String? description;
  final int? width;
  final int? height;
  int likesCount;
  int commentsCount;
  int savesCount;
  bool isLiked;
  bool isSaved;
  final CommunityUser? author;
  final DateTime createdAt;

  CommunityPost({
    required this.id,
    required this.imageUrl,
    this.thumbnailUrl,
    this.title,
    this.description,
    this.width,
    this.height,
    required this.likesCount,
    required this.commentsCount,
    required this.savesCount,
    required this.isLiked,
    required this.isSaved,
    this.author,
    required this.createdAt,
  });

  factory CommunityPost.fromJson(Map<String, dynamic> json) {
    return CommunityPost(
      id: json['id'] as int,
      imageUrl: json['imageUrl'] as String,
      thumbnailUrl: json['thumbnailUrl'] as String?,
      title: json['title'] as String?,
      description: json['description'] as String?,
      width: json['width'] as int?,
      height: json['height'] as int?,
      likesCount: json['likesCount'] as int? ?? 0,
      commentsCount: json['commentsCount'] as int? ?? 0,
      savesCount: json['savesCount'] as int? ?? 0,
      isLiked: json['isLiked'] as bool? ?? false,
      isSaved: json['isSaved'] as bool? ?? false,
      author: json['author'] != null
          ? CommunityUser.fromJson(json['author'] as Map<String, dynamic>)
          : null,
      createdAt: DateTime.tryParse(json['createdAt'] as String? ?? '') ??
          DateTime.now(),
    );
  }
}

File: awg-backend/src/entities/CommunityPost.ts
import {
    Entity,
    PrimaryGeneratedColumn,
    Column,
    CreateDateColumn,
    UpdateDateColumn,
    ManyToOne,
    JoinColumn,
    Index,
} from "typeorm";
import { User } from "./User";

@Entity("community_posts")
export class CommunityPost {
    @PrimaryGeneratedColumn()
    id!: number;

    @Column({ name: "user_id" })
    @Index()
    userId!: number;

    @ManyToOne(() => User, { onDelete: "CASCADE" })
    @JoinColumn({ name: "user_id" })
    author!: User;

    @Column({ name: "image_url", type: "text" })
    imageUrl!: string;

    @Column({ name: "thumbnail_url", type: "text", nullable: true })
    thumbnailUrl?: string;

    @Column({ length: 255, nullable: true })
    title?: string;

    @Column({ type: "text", nullable: true })
    description?: string;

    @Column({ type: "int", nullable: true })
    width?: number;

    @Column({ type: "int", nullable: true })
    height?: number;

    @Column({ name: "likes_count", default: 0 })
    likesCount!: number;

    @Column({ name: "comments_count", default: 0 })
    commentsCount!: number;

    @Column({ name: "saves_count", default: 0 })
    savesCount!: number;

    @Column({ name: "downloads_count", default: 0 })
    downloadsCount!: number;

    @Column({ name: "is_approved", default: true })
    isApproved!: boolean;

    @Column({ name: "is_reported", default: false })
    isReported!: boolean;

    @CreateDateColumn({ name: "created_at" })
    createdAt!: Date;

    @UpdateDateColumn({ name: "updated_at" })
    updatedAt!: Date;
}
